### PR TITLE
[12.0][IMP] github_connector*: Make module_paths on repository branch field functional

### PR DESCRIPTION
**Pull request by pedrobaeza**
This field allows to define where addons are located, but it's hidden right now, because
not everything is properly adjusted for paths different from root.

Making this working, there's no need to hide the field.

It includes also an extra commit that homogenize a bit the API for the methods, avoiding an extra useless argument.

cc @Tecnativa TT20539

### CODE REVIEW

File: github_connector/models/github_repository_branch.py [hunk 1]
```diff
@@ -193,11 +193,11 @@ def _get_analyzable_files(self, existing_folder):
         return res
 
     @api.model
-    def analyze_code_one(self, branch):
+    def analyze_code_one(self):
         """Overload Me in custom Module that manage Source Code analysis.
         """
         self.ensure_one()
-        path = branch.local_path
+        path = self.local_path
```
Comment (dreispt) [inline github_connector/models/github_repository_branch.py:199]: For this, you should change `@api.model` to `@api.multi`

Comment (pedrobaeza) [reply github_connector/models/github_repository_branch.py:199]: Totally, but `self` was used previously, so it's very inconsistent. Removing `@api.model`...

Comment (pedrobaeza) [reply github_connector/models/github_repository_branch.py:199]: Changed

**Review by legalsylvain — APPROVED** [review]
Thanks !

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**